x/valset-pref/validator_set_test.go

```diff
@@ -147,6 +145,244 @@ func (s *KeeperTestSuite) TestIsValidatorSetEqual() {
 	}
 }
 
+func (s *KeeperTestSuite) TestUndelegateFromValidatorSet() {
+	tests := []struct {
+		name                  string
+		delegateAmt           []osmomath.Int
+		undelegateAmt         osmomath.Int
+		noValset              bool
+		expectedUndelegateAmt []osmomath.Int
+		expectedError         bool
+	}{
+		{
```

Comment:
Isn't there another logic branch of exiting with Vratio of 1 that we could test here?

Reply:
Both of VRatio 1 and VRatio under 1 would go under same logic branch

Reply:
Both of VRatio 1 and VRatio under 1 would go under same logic branch